From bee585cf914e17807eb7ca9384a37223780a8bd3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Thu, 3 Nov 2022 06:21:38 +0100
Subject: [PATCH] Remove unnecessary escape sequences in grep expressions

grep >= 3.8 warns about stray backslashes in regular expression
patterns. These warnings are written to stderr and cause errors
when comparing the actual stderr output with the expected stderr
output.

This patch removes the stray backslashes, thus fixing the warnings.
---
 testsuite/tests/hsc2hs/Makefile                   | 2 +-
 testsuite/tests/numeric/should_run/T7014.primops  | 4 ++--
 testsuite/tests/simplCore/should_compile/Makefile | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/testsuite/tests/hsc2hs/Makefile b/testsuite/tests/hsc2hs/Makefile
index d0b2d249c3..a266d3d213 100644
--- a/testsuite/tests/hsc2hs/Makefile
+++ b/testsuite/tests/hsc2hs/Makefile
@@ -54,7 +54,7 @@ T12504:
 ifeq "$(WINDOWS)" "YES"
 	grep '{-# LINE 1 \"T12504\\\\path\\\\to\\\\$@\.hsc\" #-}' T12504/path/to/$@.hs
 else
-	grep '{-# LINE 1 \"T12504/path/to/$@\.hsc\" #-}' T12504/path/to/$@.hs
+	grep '{-# LINE 1 "T12504/path/to/$@\.hsc" #-}' T12504/path/to/$@.hs
 endif
 
 .PHONY: T15758
diff --git a/testsuite/tests/numeric/should_run/T7014.primops b/testsuite/tests/numeric/should_run/T7014.primops
index 3243666dbc..c807fb4559 100644
--- a/testsuite/tests/numeric/should_run/T7014.primops
+++ b/testsuite/tests/numeric/should_run/T7014.primops
@@ -1,8 +1,8 @@
 and#
 or#
 uncheckedShift.*#
-\+#
-\-#
++#
+-#
 \*#
 quotInt#
 remInt#
diff --git a/testsuite/tests/simplCore/should_compile/Makefile b/testsuite/tests/simplCore/should_compile/Makefile
index b1ed06bf71..6b2591f58e 100644
--- a/testsuite/tests/simplCore/should_compile/Makefile
+++ b/testsuite/tests/simplCore/should_compile/Makefile
@@ -254,7 +254,7 @@ str-rules:
 # g should have been collapsed into one defininition by CSE.
 .PHONY: T13340
 T13340:
-	'$(TEST_HC)' $(TEST_HC_OPTS) -c -O T13340.hs -ddump-simpl -dsuppress-all | grep -c '\+#'
+	'$(TEST_HC)' $(TEST_HC_OPTS) -c -O T13340.hs -ddump-simpl -dsuppress-all | grep -c '+#'
 
 
 # We expect to see all dictionaries specialized away.
